3rdParty/fuerte/src/AsioSockets.h

```diff
@@ -77,22 +91,41 @@ struct Socket<SocketType::Tcp> {
   Socket(EventLoopService&, asio_ns::io_context& ctx)
       : resolver(ctx), socket(ctx), timer(ctx) {}
 
-  ~Socket() { this->cancel(); }
+  ~Socket() { 
+    try {
+      this->cancel(); 
+    } catch (std::exception const& ex) {
+      FUERTE_LOG_ERROR << "caught exception during tcp socket shutdown: " << ex.what();
+    }
+  }
 
   template <typename F>
   void connect(detail::ConnectionConfiguration const& config, F&& done) {
-    resolveConnect(config, resolver, socket, std::forward<F>(done));
+    resolveConnect(config, resolver, socket, [this, done = std::forward<F>(done)](asio_ns::error_code ec) mutable {
+      FUERTE_LOG_DEBUG << "executing tcp connect callback, ec: " << ec.message() << ", canceled: " << this->canceled;
+      if (canceled) {
+        // cancel() was already called on this socket
+        ec = asio_ns::error::operation_aborted;
+      }
+      done(ec);
+    });
+  }
+
+  void rearm() {
+    canceled = false;
   }
 
   void cancel() {
+    canceled = true;
     try {
       timer.cancel();
       resolver.cancel();
       if (socket.is_open()) {  // non-graceful shutdown
         asio_ns::error_code ec;
         socket.close(ec);
       }
-    } catch (...) {
+    } catch (std::exception const& ex) {
+      FUERTE_LOG_ERROR << "caught exception during tcp socket cancelation: " << ex.what();
```

Comment:
By default error logging is off, so we are again silently swallowing exceptions. :/

Reply:
The only way out I see is to switch on error logging by default and leave this here as it is.